spec/controllers/users/sessions_controller_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe Users::SessionsController, type: :controller do
+  describe "POST #create" do
+    before :each do
+      @request.env["devise.mapping"] = Devise.mappings[:user]
```

Comment:
Without that line the tests don't pass and I get a tip on the command line advising me to use it. I read in Devise's GitHub page that while testing their controllers it is necessary to "tell Devise which mapping should be used before a request": https://github.com/plataformatec/devise#controller-tests